File: Algorithms/QuickSort.java
public class QuickSort {

    public static void quickSort(int arr[], int si, int ei){

        //base condition
        if (si >= ei) {
            return;
        }

        int pviot= partition(arr, si, ei);
        quickSort(arr, si, pviot-1);
        quickSort(arr, pviot+1, ei);

    }
    public static int partition(int arr[], int si, int ei){

        int pviot= arr[ei];
        int j=si-1;

        for (int i = si; i < ei; i++) {
            if ( arr[i] <= pviot) {
                j++;

                //swap 
                int temp=arr[i];
                arr[i]= arr[j];
                arr[j]=temp;
            }
        }
        j++;
        int temp= pviot;
        arr[ei]= arr[j];
        arr[j]=temp;
        return j;
        
        
    }
    public static void printArr(int arr[]){
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i]+" ");
        }
    }

    public static void main(String[] args) {
        int arr[]={4,3,6,2,1,5};
        quickSort(arr, 0, arr.length-1);
        printArr(arr);
    }
    
}

File: Algorithms/QuickSort.c
// Quick Sort
#include <stdio.h>
int partition(int arr[], int low, int high) {
    int pivot = arr[high];
    int i = low - 1;
    for (int j = low; j < high; j++) {
        if (arr[j] <= pivot) {
            i++;
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
    int tmp = arr[i + 1];
    arr[i + 1] = arr[high];
    arr[high] = tmp;
    return (i + 1);
}
void sort(int arr[], int low, int high) {
    if (low < high) {
        int p = partition(arr, low, high);
        sort(arr, low, p - 1);
        sort(arr, p + 1, high);
    }
}
int main() {
    int n;
    printf("Enter Number of Element : ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter Array Elements : \n");
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    sort(arr, 0, n - 1);
    for (int i = 0; i < n; i++)
        printf("%d ", arr[i]);
    return 0;
}